opentimelineio/opentime.py

```diff
@@ -533,29 +576,53 @@ def from_timecode(timecode_str, rate):
 
     :return: (:class:`RationalTime`) Instance for the timecode provided.
     """
+    dropframe = False
+
+    if not isinstance(rate, (int, float)):
+        raise ValueError(
+                "rate must be <float> or <int> not {t}".format(t=type(rate))
+                )
 
+    # Make sure rate is float
+    rate = float(rate)
+
+    # Check if rate is drop frame
+    if not rate.is_integer():
+        # 23.98 is not considered drop frame
+        if not math.ceil(rate) == 24.0:
```

Comment:
If its a fixed set of drop frame timecode that are valid, I'd rather test membership in that set rather than this ceil approach - something like `VALID_DROPFRAME_TIMECODE_RATES` so you could say `is_dropframe = rate in VALID_DROPFRAME_TIMECODE_RATES`

Reply:
Agree.